api/src/main/java/org/apache/iceberg/ManifestFile.java

```diff
@@ -42,7 +42,10 @@
           required(509, "contains_null", Types.BooleanType.get()),
           optional(510, "lower_bound", Types.BinaryType.get()), // null if no non-null values
           optional(511, "upper_bound", Types.BinaryType.get())
-      ))));
+      ))),
+      optional(512, "added_rows_count", Types.LongType.get()),
+      optional(513, "existing_rows_count", Types.LongType.get()),
+      optional(514, "deleted_rows_count", Types.LongType.get()));
```

Comment:
Can we add these up by the data files counts?

Reply:
@rdblue, not sure I got. Do you mean whether we can avoid storing these and add them up?

Reply:
No, I meant that we don't need to add these at the end of the schema. We can add them up by the file count columns, like you do with all of the accessor methods.

Reply:
Yes, let me update this.

Reply:
Done.

Reply:
Hm, reordering actually led to failures while writing manifest lists.

Reply:
@rdblue, do we want to build something as `ProjectionDatumReader` for the write side?

Reply:
@rdblue, I reverted this change. I propose to merge it as is and create a follow-up issue to implement writers that take into account field ids.

Reply:
Writers don't use field IDs?

Reply:
@rdblue, yes, `GenericAvroWriter` doesn't. I think `SparkAvroWriter` does respect field ids.